src/org/labkey/test/tests/DomainDesignerTest.java

```diff
@@ -305,6 +305,13 @@ public void testBlankNameFieldOnAddedField() throws Exception
         assertTrue("expect error to contain [Please provide a name for each field.] but was[" + hasNoNameError + "]",
                 hasNoNameError.contains("Please provide a name for each field."));
 
+        String warningfieldMessage = noNameRow.setName("&totally not a sketchy sql injection attack")
+                .waitForWarning()
+                .detailsMessage();
+        String expectedWarning = "New field. Warning: SQL queries, R scripts, and other code are easiest to write when field names only contain combination of letters, numbers, and underscores, and start with a letter or underscore";
+        assertTrue("expect error to contain [" + expectedWarning + "] but was[" + warningfieldMessage + "]",
+                warningfieldMessage.contains(expectedWarning));
+
         domainDesignerPage.clickCancelAndDiscardChanges();
```

Comment:
Save domain to make sure these are actually non-blocking warnings.

Reply:
these are blocking warnings.  It won't enable the 'save' button if your field name is [&totally not a sketchy sql injection attack].  Can't save it that way.